src/coreclr/nativeaot/Runtime.Base/src/System/Runtime/TypeCast.cs

```diff
@@ -37,21 +37,145 @@ internal static class TypeCast
         [Flags]
         internal enum AssignmentVariation
         {
-            Normal = 0,
+            /// <summary>
+            /// Conversion from an object. In the terminology of ECMA335 the relationship is called "compatible-with".
+            /// This is the relation used by castclass and isinst (III.4.3).
+            /// Value types are compatible with Object, ValueType and Enum (if applicable) and implemented interfaces.
+            /// </summary>
+            BoxedSource = 0,
```

Comment:
I have changed the enum here. Casting from objects is a much more common operation than reasoning about convertibility of types. When converting from objects the source is always in an object/boxed form, so it would be nice to have `variation == 0` for that case.